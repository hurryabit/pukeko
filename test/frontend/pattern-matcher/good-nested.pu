data List a =
  | Nil
  | Cons a (List a)
last : a -> List a -> a
last x0 xs =
  match xs with
  | Nil                -> x0
  | Cons x Nil         -> x
  | Cons _ (Cons x xs) -> last x xs
