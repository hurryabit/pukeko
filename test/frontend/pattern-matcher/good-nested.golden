data List a =
       | Nil
       | Cons a (List a)
last :: forall a. a -> List a -> a =
  \@a ->
    \(x0 :: a) ->
      \(xs :: List a) ->
        match xs with
        | Nil -> x0
        | Cons pm$1 pm$2 ->
          match pm$2 with
          | Nil -> pm$1
          | Cons x xs -> last @a x xs
